pkg/cfn/manager/waiters.go

```diff
@@ -0,0 +1,182 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+
+	"github.com/kubicorn/kubicorn/pkg/logger"
+
+	"github.com/weaveworks/eksctl/pkg/cfn/builder"
+)
+
+// cloudformation.WaitUntilStackCreateComplete doesn't detect in-progress status early enough,
+// so this is custom version that is more suitable for our use, as there is no way to add any
+// custom acceptors
```

Comment:
Arguably, this new code is probably as many lines (or more), however it's really not as complicated as what we've had, we mostly define structs here, and call the AWS SDK to do polling (only if we didn't have to poll...).

Fun fact: CloudFormation API is actually XML... why cannot we write templates in XML with XSLT and all that? 🤔 😉